migration-engine/connectors/sql-migration-connector/src/sql_renderer/postgres_renderer.rs

```diff
@@ -298,7 +314,21 @@ impl SqlRenderer for PostgresFlavour {
         .to_string()
     }
 
-    fn render_create_table_as(&self, table: TableWalker<'_>, table_name: &str) -> String {
+    fn render_create_namespace(&self, ns: sql_schema_describer::NamespaceWalker<'_>) -> String {
+        format!("CREATE SCHEMA IF NOT EXISTS {}", Quoted::postgres_ident(ns.name()))
```

Comment:
Do we have some stance on when to use `If Not Exists` and when not?

Reply:
Not an explicit, written one. My stance so far has been "never use IF NOT EXISTS, we should always know if something exists or not in diffing". We have a first exception in the postgres extensions proposal, motivated by not wanting to make the feature a breaking change.

In this PR's case, we are not diffing anything yet, we always return `CREATE SCHEMA IF NOT EXISTS _` for all schemas in all migrations when the preview feature is enabled and schemas are used. It's not what we want when we seriously implement the feature, I expect we'll remove the IF NOT EXISTS then because we'll have proper diffing of schemas.

Reply:
I think I'll write a short sentence on the IF NOT EXISTS topic in the migrate FAQ.

Reply:
done in [`5f17521` (#3128)](https://github.com/prisma/prisma-engines/pull/3128/commits/5f17521f2121637c62ebf8df607c933f221cfa82)